syntax = "proto3";

service UserService {
  rpc GetUsers (Empty) returns (UsersList);
  rpc AddUser (NewUser) returns (User);
  rpc GetUserByOS(OSRequest) returns (UsersList);
}

message Empty {}

message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string preferred_os = 4;
}

message NewUser {
  string name = 1;
  string email = 2;
  string preferred_os = 3;
}

message UsersList {
  repeated User users = 1;
}

message OSRequest{
    string preferred_os = 1;
}